### DISCUSSION TITLE
How to get full matrix as return value from eval()?

### DISCUSSION THREAD

**Discussion by iandanforth** [josdejong/mathjs#3070] [Q&A]
In contrast to Matlab, for example, if I assign a value to a portion of a matrix I only get back that portion. I'd like to do something like

```
A[1,:] ./ 2;
```

And get back the whole matrix, not just the first row.

Is this possible?

**Comment by josdejong** [comment]
Thanks for bringing this up Ian. This behavior looks odd: replacing a subset of a matrix returns only the affected part of the matrix. This behavior may stem from assignments like A[1,1] = 4 which returns `4` (like many programming languages), but I would not expect this behavior in an operation.

To get back the whole matrix, you will simply have to execute a second expression `A` to get the whole matrix.

**Comment by gwhitney** [comment]
(A) I am going to move this to questions.

(B) I am in favor of mathjs' current behavior. Note that `A[1, :] ./ 2` does not (perhaps it once did, but if so, does no longer) modify A. So in my reading, `A[1,:]` represents  a submatrix (the first row, but as a 1xn matrix) of A, which is then elementwise divided by 2, and you get just that 1xn result. I think it would be very surprising if you got all of A as a result of this operation.
